### PR TITLE
Refactor algorithm page

### LINKED ISSUE #16: Refactor client code to meet design and standard
Currently our client app code is hard to follow, lacks docstring, or variable naming does not match standard. We should refactor the client-side code according to design outlined in https://github.com/mytnguyen26/algo-shadow/blob/dev/documents/software_design_document.pdf 

**Describe the solution you'd like**
- apply automatic linter to code every commit to make sure our code meets standard
- refactor algorithm logic and animation module for reusability
- add doc strings to all method and modules

### PR DESCRIPTION
**Pull request by mytnguyen26**
# Description
This PR introduces the following changes

## I. Refactor Algorithm Page and solvers, and create modules:
- `GraphRenderer`: responsible for common layout and button actions across all algorithms route.
- `AlgorithmSolver`: responsible for executing and create data structure for each type of algorithms

## II. Styling and Naming Standard:
- refactor variable names and module name to follow JavaScript standard:
- variable names should be descriptive. For example: `var c` should be `var circle`, `var p` should be `var line`.
-

### III. Docstring for Module and Methods:

- All functions, class, and module should have Docstrings that explains what each of them does.
- Only English should be used for comments

This PR closes #16 

## Checklist

- [ ] This PR can be reviewed in under 30 minutes
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have assigned reviewers to this PR.

### CODE REVIEW

**Review by benbencfy — APPROVED** [review]

**Comment by mytnguyen26** [comment]
Close, since we merged #30
